logs/logs.go

```diff
@@ -0,0 +1,53 @@
+// Package logs provides the standard interface and handler for OpenFaaS providers to expose function logs.
+//
+package logs
+
+import (
+	"fmt"
+	"time"
+)
+
+// Request is the query to return the function logs.
+type Request struct {
+	// Name is the function name and is required
+	Name string `json:"name"`
+	// Instance is the optional container name, that allows you to request logs from a specific function instance
+	Instance string `json:"instance"`
+	// Since is the optional datetime value to start the logs from
+	Since *time.Time `json:"since"`
+	// Limit sets the maximum number of log messages to return, <=0 means unlimited
+	Limit int `json:"limit"`
+	// Follow is allows the user to request a stream of logs
+	Follow bool `json:"follow"`
+	// Pattern is an optional regexp value to filter the log messages
+	Pattern *string `json:"pattern"`
```

Comment:
I didn't think this would be in scope for v1. Did this feature creep in from the interfaces in Kubernetes/Swarm?

Reply:
A couple of reasons, 
1. I wanted to ensure that it could be implemented, 
2. it didn't take very long, 
3. and to ensure consistent logging behavior i wanted to emphasize it being done in the provider instead of trying to pass filters to the various providers in some way